src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/ForeignLinker.java

```diff
@@ -32,7 +32,7 @@
  * This class can be used to link native functions as a {@link MethodHandle}, or to link Java
  * methods as a native function pointer (modelled as a {@link MemorySegment}).
  *
- * Instances of this interface can be obtained for instance by calling {@link CSupport#getSystemLinker()}
+ * Instances of this interface can be obtained for instance by calling {@link CLinker#getSystemLinker()}
```

Comment:
This comments should says something else - e.g. ```
Subclasses of this interface provide support for different linking strategies. For instance, the {@link CLinker} interface defines a foreign linker which supports the C ABI.```

Reply:
We are in a situation where we have an interface and only one implementation (`CLinker`) - should we just have the latter, and maybe reserve the room to add the general interface at a later time, as we have more implementations? @PaulSandoz - what do you think?

We started off in this direction because we thought that it might have been useful to have different places where to document general foreign linker functionality - as opposed to functionalities that are only applicable to the C linker - but the interface is so general that I'm not sure exactly what we can say at this level.

Reply:
We also need a comment about sealedness and immutability here

Reply:
> We are in a situation where we have an interface and only one implementation (`CLinker`) - should we just have the latter, and maybe reserve the room to add the general interface at a later time, as we have more implementations? @PaulSandoz - what do you think?
> 

Now that i have looked more closely `ForeignLinker` does not seem weighty enough, even though the concepts of `downcall` and `upcall` are important. I suspect a common super interface could be defined if another kind of linker becomes a realization, and even then it's not clear to me that would be useful given the following on the `CLinker`:

> * Clients of this API should build function descriptors using the predefined memory layout constants
> * (based on a subset of the built-in types provided by the C language), found in this interface;

which indicates some tighter coupling.

Reply:
I agree. Thanks for the fresh perspective :)

CLinker was created as a place to put the C specific documentation, but there's probably no need for different linkers to have a common supertype (ForeignLinker). Maybe general doc about what 'linking' means can be put in package-info?

Reply:
I think putting general linking info in the package-level doc sounds totally the right thing to do.